src/Components/Endpoints/src/Results/RazorComponentResultExecutor.cs

```diff
@@ -55,7 +57,10 @@ public static Task ExecuteAsync(HttpContext httpContext, RazorComponentResult re
                 { nameof(RazorComponentEndpointHost.ComponentParameters), componentParameters },
             });
 
-            await using var writer = CreateResponseWriter(httpContext.Response.Body);
+            // Matches MVC's MemoryPoolHttpResponseStreamWriterFactory.DefaultBufferSize
+            var defaultBufferSize = 16 * 1024;
+            await using var writer = new HttpResponseStreamWriter(httpContext.Response.Body, Encoding.UTF8, defaultBufferSize, ArrayPool<byte>.Shared, ArrayPool<char>.Shared);
+            using var bufferWriter = new BufferedTextWriter(writer);
```

Comment:
This is separate from the rest of the PR, but I realised that `RazorComponentResultExecutor` is inconsistent with `RazorComponentEndpointInvoker`.

It's dangerous that we have two different implementations as they could be out of sync and not support the same features. And here in fact one of them was not using the new buffering solution. However I don't want to unpick all the complexities of unifying these code paths within this PR (and I think we likely should leave that for .NET 9), so I'm just changing `RazorComponentResultExecutor` to use the same buffering technique as `RazorComponentEndpointInvoker`. Longer term, we should unify the code.

Note that fixing this forced me to make some other unit test updates (see `WaitForContentWrittenAsync` in this PR), as the unit tests were relying on content writes being completely synchronous.